### ISSUE TITLE
Yearly-repeating dates from synced Nextcloud-calendar within years [1967,2001] and without year are missing

### ISSUE THREAD

**Issue by zifredder** [SimpleMobileTools/Simple-Calendar#776]
TL; DR: Yearly-repeating events starting between 1967 and 2001 that are synced from Nextcloud are missing in Simple Calendar, while locally stored events starting in this time frame are displayed.

Might be related to #768.

I manage my contacts via a Nextcloud instance where birthday events are automatically created from the contacts in a separate birthday calendar which I sync to my phone using DAVx5. I know I could use the add-contact-birthdays function of Simple Calendar, but I would like to keep the birthday event management on the Nextcloud side.

Since a while -- I don't know when exactly this occurred for the first time (I'm running version 6.2.2, I have not tried to reproduce this with older versions) -- most of the birthday dates are missing in the app. Not all events, however: My debugging skills are weak, but I think I have traced the source of this issue to the year of the contact's birthday. Contacts where the year is not specified (e. g. Google's contact manager app allows this) and contacts where the birthday year is between 1967 and 2001 (including) are not shown by Simple Calendar.
As I explored further, I found out that this affects yearly-repeating events that start within this time frame in general, independent from my birthday calendar. Such events created elsewhere and then synced are not displayed. When I create such an event using Simple Calendar, the event is shown, but only until I sync the calendar, after which it disappears in the app.
The events are shown in the Nextcloud calendar, by other apps on my phone, such as Google's calendar app, and by software on my PC, such as Mozilla Thunderbird's calendar.

If I create a yearly-repeating event in this time frame locally on my phone, it will not disappear. 
This is the data of a local event, exported via Simple Calendar, being displayed in the app:

```
BEGIN:VCALENDAR
PRODID:-//Simple Mobile Tools//NONSGML Event Calendar//EN
VERSION:2.0
BEGIN:VEVENT
SUMMARY:Testrep2
UID:Caldav-12-1253
CATEGORY_COLOR:-2937041
CATEGORIES:Regular event
LAST-MODIFIED:20190206T164201Z
DTSTART;VALUE=DATE:20000207
DTEND;VALUE=DATE:20000208
STATUS:CONFIRMED
RRULE:FREQ=YEARLY;INTERVAL=1;BYMONTH=2
END:VEVENT
END:VCALENDAR
```

This is the data of an event created outside Simple Calendar, synced via Nextcloud, exported via Nextcloud, not being displayed in Simple Calendar:

```
BEGIN:VCALENDAR
PRODID:-//Mozilla.org/NONSGML Mozilla Calendar V1.1//EN
VERSION:2.0
BEGIN:VEVENT
CREATED:20190206T162308Z
LAST-MODIFIED:20190206T162513Z
DTSTAMP:20190206T162513Z
UID:4a0e6616-60ce-4580-a477-d5c14b524b6f
SUMMARY:testrepyear
RRULE:FREQ=YEARLY
DTSTART;VALUE=DATE:20000206
DTEND;VALUE=DATE:20000207
TRANSP:TRANSPARENT
SEQUENCE:1
X-MOZ-GENERATION:1
END:VEVENT
END:VCALENDAR
```

And this is the data of an event created outside Simple Calendar, synced via Nextcloud, being displayed in and exported via Simple Calendar:
```
BEGIN:VCALENDAR
PRODID:-//Simple Mobile Tools//NONSGML Event Calendar//EN
VERSION:2.0
BEGIN:VEVENT
SUMMARY:testrepyear
UID:Caldav-12-1254
CATEGORY_COLOR:-16744193
CATEGORIES:Personal
LAST-MODIFIED:19700101T000000Z
DTSTART;VALUE=DATE:20060205
DTEND;VALUE=DATE:20060206
STATUS:CONFIRMED
RRULE:FREQ=YEARLY;INTERVAL=1;BYMONTH=2
END:VEVENT
END:VCALENDAR
```

I basically don't know anything about this topic, so the differences in the data of these events tell me nothing -- to me, the only difference between the first (displayed) and the last (missing) is the date.

Since locally stored events are displayed, it seems like a bug related to Nextcloud. Is this reproducible by anyone? The fact that it seems to be related to a particular time frame of years seems very odd to me.

If I am told how, I will gladly debug further.

**Comment by tibbi** [comment]
date handling is not done ideally, it could glitch before 1970. Can you check if that is the case? Is the issue related to dates before 1970?

**Comment by zifredder** [comment]
I narrowed down the year interval in which the events are not displayed by trial and error, and ended up with the observation that events starting in 1966 and before are displayed and events starting in 1967 and after are not. This goes up to 2001, where events are missing, but starting with 2002, events are displayed again.

I have now realized that it has nothing to do with the repetition-property of the events, but just with the year. If I create an event on, e. g. 01/01/2000, and sync the calendar with my Nextcloud, it will not be displayed in Simple Calendar. Can you reproduce this? Everything works with locally stored events, though.

Obviously, this is not relevant for my day-to-day calendar events, but for birthday events, it is.

**Comment by tibbi** [comment]
weird.. Im not working on the calendar now, will take a look at it someday later. Thanks

**Comment by stefanbuettner** [comment]
Hey there,

I stumbled over this issue while investigating why SimpleCalendar doesn't show all my contact birthdays synced from nextcloud.

I can confirm, that
- The default calendar app does show all birthdays.
- Birthdays of people being born outside the [1967, 2001] interval are shown whereas birthdays of people born within this interval aren't.
- An event *E* created within SimpleCalendar on e.g. 2000-08-17 is shown in SimpleCalendar and the default calendar app as long as it is not synced to Nextcloud.
- The event *E* synced to Nextcloud vanishes in the SimpleCalendar.

Cheers,
Stefan

**Comment by tibbi** [comment]
do you have those old events shown in any calendar whatsoever? Ive just created an event on 01/01/2000 and it is shown properly in NextCloud, but not even Google Calendar shows it. Maybe sync adapters just ignore so old events.

**Comment by tibbi** [comment]
aha once I make it repeatable it shows up in google calendar already, just not in Simple Calendar. So there must be some issue.

**Comment by tibbi** [comment]
so it was obviously my foolish mistake. Should be fixed in the next version 6.3.0 released soon. Thanks
